src/Stratis.Bitcoin/P2P/Protocol/Behaviors/EnforcePeerVersionCheckBehavior.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities;
+using TracerAttributes;
+
+namespace Stratis.Bitcoin.P2P.Protocol.Behaviors
+{
+    /// <summary>
+    /// Sets the minimum supported client version <see cref="this.NodeSettings.MinProtocolVersion"> to <see cref="this.Network.Consensus.Options.EnforcedMinProtocolVersion">
+    /// based on the predefined block height <see cref="this.Network.Consensus.Options.EnforceMinProtocolVersionAtBlockHeight">.
+    /// Once the new minimum supported client version is changed all existing peer connections will be dropped upon the first received message from outdated client.
+    /// </summary>
+    public class EnforcePeerVersionCheckBehavior : NetworkPeerBehavior
```

Comment:
It would be very beneficial to have a unit test for this class, specifically testing logic in `OnMessageReceivedAsync`. Just in case there is a bug of we change the approach slightly, this will be acting as a guard.

Reply:
Added unit tests